pkg/controllers/secretstore/common.go

```diff
@@ -93,8 +97,8 @@ func validateStore(ctx context.Context, namespace string, store esapi.GenericSto
 	}
 	defer cl.Close(ctx)
 
-	err = cl.Validate()
-	if err != nil {
+	state, err := cl.Validate()
+	if err != nil && !(state == esapi.ClientStateUnknown) {
```

Comment:
Do we expect `esapi.ClientStateUnReady` to also error out?

If so, we might need to add this condition here as well.

Reply:
I think this is fine as it is: when unknown -> error is skipped and it can be considered ready